clang/test/ParserSYCL/unique_stable_name.cpp

```diff
@@ -0,0 +1,33 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -Wno-unused %s
```

Comment:
Is the `-Wno-unused` needed?

Reply:
For all of the tests that pass or fail in this file, you should have a comment that explains why the test behaves the way it does (right now the comments just describe what is being tested but nothing about the expected outcomes).

Reply:
`-Wno-unused` is because I am several 
`'warning' diagnostics seen but not expected`:
  File /clang/test/ParserSYCL/unique_stable_name.cpp Line 14: `expression result unused`

Reply:
Added comments with expectations for test cases